src/status_im/chat/models.cljs

```diff
@@ -265,13 +265,21 @@
 (fx/defn start-public-chat
   "Starts a new public chat"
   [cofx topic {:keys [dont-navigate?] :as opts}]
-  (fx/merge cofx
-            (add-public-chat topic)
-            #(when-not dont-navigate?
-               (navigate-to-chat % topic opts))
-            (public-chat/join-public-chat topic)
-            (when platform/desktop?
-              (desktop.events/change-tab :home))))
+  (let [chats           (into []
+                              (comp (filter :is-active)
+                                    (filter :public?)
+                                    (filter #(= topic (:chat-id %))))
+                              (vals (get-in cofx [:db :chats])))
+        already-active? (seq chats)]
+    (if already-active?
+      (if (not dont-navigate?) (navigate-to-chat cofx topic opts) cofx)
```

Comment:
you never want to return the cofx as effects 
the if here could be a when

Reply:
got it

Reply:
> @bitsikka why did you use a negative predicate `dont-sync?` instead of `sync?`, I think even `dont-navigate` would be better as `navigate?` but doesn't need to be fixed in this PR. It's weird to have negative predicates and lead to double negations in the code which can be confusing

It is weird indeed - I was just following `dont-navigate?` which existed before, and it kind-of made sense when coming from dispatch site. `dont-sync?` is only ever set in `status-im.pairing.core/sync-public-chat`, everywhere else it is ignored hence nil/false implicitly by-default.